index.js

```diff
@@ -2,17 +2,18 @@
 const fs = require('fs');
 const stream = require('stream');
 const zlib = require('zlib');
+const {promisify} = require('util');
 const duplexer = require('duplexer');
-const pify = require('pify');
 
-const getOptions = options => Object.assign({level: 9}, options);
+const getOptions = options => ({level: 9, ...options});
 
-module.exports = (input, options) => {
+module.exports = async (input, options) => {
 	if (!input) {
 		return Promise.resolve(0);
 	}
 
-	return pify(zlib.gzip)(input, getOptions(options)).then(data => data.length).catch(_ => 0);
+	const data = await promisify(zlib.gzip)(input, getOptions(options));
```

Comment:
It's generally better to do the promisification at the top-level if it doesn't require any local state, so it doesn't have to be done for each function invocation.